### PR TITLE
Expose getAppPath to public API

### PR DESCRIPTION
**Pull request by juliusknorr**
As discussed with @rullzer in #840 

I'm not sure if this is fine for now or if the whole logic of getAppPath should be moved from OC_App to the AppManager.

### CODE REVIEW

**Review by LukasReschke — APPROVED** [review]
LGTM

**Comment by LukasReschke** [comment]
LGTM

File: lib/private/App/AppManager.php [hunk 1]
```diff
@@ -266,6 +266,21 @@ public function disableApp($appId) {
...
+	 * Get the directory for the given app.
+	 *
+	 * @param string $appId
+	 * @return string
+	 * @throws \Exception if app folder can't be found
+	 */
+	public function getAppPath($appId) {
+		$appPath = \OC_App::getAppPath($appId);
+		if($appPath === false) {
+			throw new \Exception('Could not find path for ' . $appId);
```
Comment (nickvergessen) [inline lib/private/App/AppManager.php]: Should throw a more specific exception? Having to catch the generic `\Exception` is always a pain.

**Comment by nickvergessen** [comment]
> I'm not sure if this is fine for now or if the whole logic of getAppPath should be moved from OC_App to the AppManager.

Fine for now.

**Comment by juliushaertl** [comment]
@nickvergessen Ok, moved to OCP\Files\NotFoundException.

File: lib/public/App/IAppManager.php [hunk 2]
```diff
@@ -79,6 +79,15 @@ public function enableAppForGroups($appId, $groups);
 	public function disableApp($appId);
 
 	/**
+	 * Get the directory for the given app.
+	 *
+	 * @param string $appId
+	 * @return string
+	 * @since 11.0.0
```
Comment (nickvergessen) [inline lib/public/App/IAppManager.php]: missing `@throws`

File: lib/private/App/AppManager.php [hunk 3]
```diff
@@ -33,6 +33,7 @@
 
 use OCP\App\IAppManager;
 use OCP\App\ManagerEvent;
+use OCP\Files\NotFoundException;
```
Comment (nickvergessen) [inline lib/private/App/AppManager.php]: Create one in `OCP\App` ?

**Review by nickvergessen — DISMISSED** [review]

**Comment by juliushaertl** [comment]
@nickvergessen I've added a AppPathNotFoundException. Please review.

**Comment by MorrisJobke** [comment]
:+1:

**Comment by juliushaertl** [comment]
Guess it could be time to move after 7 years

**Pull request merged**